### ISSUE TITLE
MD013: False positive on a single url with a manual line break

### ISSUE THREAD

**Issue by wildwestrom** [DavidAnson/markdownlint#584]
So I'm using `markdownlint` for the first time, and I think the behavior should be slightly different.

This does not trigger the lint.
`![linkylink](https://example.com/a-really-really-long-link-that-is-longer-than-80-chars)`
This does. I still think it shouldn't trigger the lint though, I'm just trying to do a manual line break.
(representing spaces as underscores)
`![linkylink](https://example.com/a-really-really-long-link-that-is-longer-than-80-chars)__`

Currently I'm doing:
```md
![linkylink](https://example.com/a-really-really-long-link-that-is-longer-than-80-chars)<!-- markdownlint-disable-line MD013 -->__
```

**Comment by DavidAnson** [comment]
While your example does produce MD013, it is also missing a dash in the URL, producing MD034. Adding that missing dash addresses both issues and no in-line comment is needed.

Here's an example: https://dlaa.me/markdownlint/#%25m%23%20Issue%20584%0A%0A%3C!--%20markdownlint-enable%20MD013%20--%3E%0A%0A!%5Bformat%5D(https%3A%2F%2Fexample.com%2Fa-really-really-long-link-that-is-longer%20than-80-chars)%20%20%0A%0A!%5Bformat%5D(https%3A%2F%2Fexample.com%2Fa-really-really-long-link-that-is-longer-than-80-chars)%20%20%0A

**Comment by wildwestrom** [comment]
My bad, I'll update my example.

**Comment by wildwestrom** [comment]
Weird, it's true, the [demo](https://dlaa.me/markdownlint/#%25m%23%20Death%20Calendar%0A%0A!%5Bdocs%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fdocs.yml%2Fbadge.svg)%0A!%5Btests%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Ftests.yml%2Fbadge.svg)%0A!%5Bformat%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fformat.yml%2Fbadge.svg)%20%20%0A!%5Blint%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Flint.yml%2Fbadge.svg)%0A!%5Bbuild%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fbuild.yml%2Fbadge.svg)%0A!%5Bpackage%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fpackage.yml%2Fbadge.svg)%0A%0AEver%20wonder%20how%20many%20days%20you%20have%20left%20to%20live%3F%20%20%0ADeath%20Calendar%20can%20show%20you%20at%20a%20glance.%0A%0AThe%20main%20feature%20is%20its%20SVG%20calendar%20generator.%20It's%20meant%20to%20be%20used%20as%20a%0Areminder%20of%20your%20mortality%2C%20so%20that%20you%20can%20make%20the%20most%20of%20your%20limited%20time%0Aon%20Earth.%0A%0AThis%20was%20100%25%20inspired%20by%20the%0A%5Bwaitbutwhy.com%5D(https%3A%2F%2Fwaitbutwhy.com%2F2014%2F05%2Flife-weeks.html)%20article%2C%0Avisualizing%20human%20life%20using%20various%20charts.%0A%0A%23%23%20Installation%0A%0AMake%20sure%20you%20have%20a%20working%20rust%20toolchain.%0A%0A%60%60%60shell%0A%23%20Shallow%20clone%20the%20repository%0Agit%20clone%20--depth%201%20https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%0A%23%20Move%20into%20it%0Acd%20death-calendar%0A%23%20Compile%20it%0Acargo%20build%20--release%0A%23%20Now%20copy%20the%20newly%20compiled%20file%20to%20a%20location%20on%20your%20%24PATH%0Acp%20target%2Frelease%2Fdeath-calendar%20%5BDESTINATION%5D%0A%60%60%60%0A%0A%23%23%20Goals%0A%0A-%20Make%20an%20easy%20installer%0A-%20Make%20it%20so%20the%20calendar%20updates%20every%20day%20automatically%0A-%20Make%20a%20graphical%20interface%20for%20customizing%20the%20look%20of%20the%20calendar%0A-%20Should%20work%20on%20multiple%20platforms%0A%20%20-%20Linux%0A%20%20-%20MacOS%0A%20%20-%20Windows%0A%20%20-%20Android%0A%20%20-%20iOS%0A%0A%23%23%23%20New%20Calendar%20Render%20Ideas%0A%0A-%20Make%20a%20spiral%20calendar%0A-%20Make%20a%20logarithmic%20calendar%0A-%20Represent%20your%20time%20as%20an%20hourglass%0A%0A%23%23%20License%0A%0ADeath%20Calendar%3A%20Calculate%20how%20much%20time%20you%20have%20until%20your%20ultimate%20demise.%20%20%0ACopyright%20%C2%A9%202021%20Christian%20Westrom%0A%0AThis%20program%20is%20free%20software%3A%20you%20can%20redistribute%20it%20and%2For%20modify%20it%20under%0Athe%20terms%20of%20the%20GNU%20Affero%20General%20Public%20License%20as%20published%20by%20the%20Free%0ASoftware%20Foundation%2C%20either%20version%203%20of%20the%20License%2C%20or%20(at%20your%20option)%20any%0Alater%20version.%0A%0AThis%20program%20is%20distributed%20in%20the%20hope%20that%20it%20will%20be%20useful%2C%20but%20WITHOUT%20ANY%0AWARRANTY%3B%20without%20even%20the%20implied%20warranty%20of%20MERCHANTABILITY%20or%20FITNESS%20FOR%20A%0APARTICULAR%20PURPOSE.%20See%20the%20GNU%20Affero%20General%20Public%20License%20for%20more%20details.%0A%0AYou%20should%20have%20received%20a%20copy%20of%20the%20GNU%20Affero%20General%20Public%20License%20along%0Awith%20this%20program.%20If%20not%2C%20see%20%3Chttps%3A%2F%2Fwww.gnu.org%2Flicenses%2F%3E.%0A) doesn't give me an error. However when I use the cli tool it does.

**Comment by DavidAnson** [comment]
The demo disables MD013 by default which is why you may have noticed I enabled it in my link: `<!-- markdownlint-enable MD013 -->`. If I do that in your latest example, I do see a single warning for the third image but that goes away when I remove the trailing two spaces you are using to introduce the line break. Without analyzing this too much, I think trailing spaces are within the spirit of this rule and so I will look at weather that pattern should be allowed. For now, you could disable the rule as you showed in the original post or maybe more conveniently via `<!-- markdownlint-disable-next-line MD013 -->`.

**Comment by wildwestrom** [comment]
Gotcha, thanks for clarifying.

For the record, using `<!-- markdownlint-disable-next-line MD013 -->` causes an unwanted line break.

**Comment by DavidAnson** [comment]
After revisiting this, I am **not** going to make a change, as I think the current behavior is reasonable. The intent behind allowing URLs to exceed the desired line length, is because they cannot easily be broken up. In this case, the two spaces at the end to produce a line break could be replaced by an actual line break which is more explicit and less likely to be overlooked way off on the right edge of the screen.

[Link to above demo with that change](https://dlaa.me/markdownlint/#%25m%3C!--%20markdownlint-enable%20MD013%20--%3E%0A%0A%23%20Death%20Calendar%0A%0A!%5Bdocs%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fdocs.yml%2Fbadge.svg)%0A!%5Btests%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Ftests.yml%2Fbadge.svg)%0A!%5Bformat%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fformat.yml%2Fbadge.svg)%0A%0A!%5Blint%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Flint.yml%2Fbadge.svg)%0A!%5Bbuild%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fbuild.yml%2Fbadge.svg)%0A!%5Bpackage%5D(https%3A%2F%2Fgithub.com%2Fwildwestrom%2Fdeath-calendar%2Factions%2Fworkflows%2Fpackage.yml%2Fbadge.svg)%0A)
